### PR TITLE
Issue #729 fix empty search field

### PR DESCRIPTION
**Pull request by ejnshtein**
Currently search field allows to write empty symbols (space) and search it which brakes code with 404 as shown here.
This solution should resolve issue with search field and search preview.

### CODE REVIEW

File: src/app/style/base/_keyframes.scss [hunk 1]
```diff
@@ -9,7 +9,7 @@
  * @link https://github.com/scandipwa/base-theme
  */
 
-@keyframes placeholderAnimation {
+@keyframes placeholderanimation {
```
Comment (alfredsgenkins) [inline src/app/style/base/_keyframes.scss:12]: But... Why? :d

File: src/app/component/SearchField/SearchField.component.js [hunk 2]
```diff
@@ -64,10 +64,10 @@ class SearchField extends PureComponent {
     }
 
     onSearchEnterPress = (e) => {
-        if (e.key === 'Enter') {
-            const { searchCriteria, hideActiveOverlay, onSearchBarChange } = this.props;
-            const search = searchCriteria.replace(/\s\s+/g, '%20');
-
+        const { searchCriteria, hideActiveOverlay, onSearchBarChange } = this.props;
```
Comment (alfredsgenkins) [inline src/app/component/SearchField/SearchField.component.js:67]: Put every value on separate line.

File: src/app/component/SearchField/SearchField.component.js [hunk 3]
```diff
@@ -64,10 +64,10 @@ class SearchField extends PureComponent {
...
+        const search = searchCriteria.trim().replace(/\s\s+/g, '%20');
+        const trimmedSearch = searchCriteria.trim();
```
Comment (alfredsgenkins) [inline src/app/component/SearchField/SearchField.component.js:69]: Separate this from the bellow if with new-line

File: src/app/component/SearchField/SearchField.component.js [hunk 4]
```diff
@@ -93,9 +93,11 @@ class SearchField extends PureComponent {
     handleChange = (e) => {
         const { target: { value } } = e;
         const { onSearchBarChange } = this.props;
-        onSearchBarChange(e);
+        const trimmedValue = value.trim();
+
+        onSearchBarChange({ target: { value: trimmedValue } });
```
Comment (alfredsgenkins) [inline src/app/component/SearchField/SearchField.component.js:98]: Wow that's very unsafe, do not fake events like this. Please change the `onSearchBarChange` to trim, rather then faking input.

File: src/app/style/base/_keyframes.scss (hunk 1 above)
Comment (ejnshtein) [reply src/app/style/base/_keyframes.scss:12]: css lint was failing

File: src/app/component/SearchField/SearchField.component.js (hunk 4 above)
Comment (ejnshtein) [reply src/app/component/SearchField/SearchField.component.js:98]: trimStart should be placed in search preview then

**Pull request closed without merging**